generators/repo/src/commands/generate.ts

```diff
@@ -375,53 +378,51 @@ export class GenerateCommand implements RepomanCommand {
+        const globOptions: IOptions = { 
+            cwd: this.generatePath, 
+            ignore: rule.ignore,
+            matchBase: true 
+        };
+        const patterns = rule.patterns ?? [];
+        for (const pattern of patterns) {
+            const files = await getGlobFiles(pattern, globOptions);
+            for (const filePath of files) {
+                await this.rewritePath(rule, filePath);
             }
+        }
+    }
 
+    private rewritePath = async(rule: RewriteRule, filePath: string) => {
+        const destFilePath = path.join(this.generatePath, filePath);
+        const destFolderPath = path.dirname(destFilePath);
+        const buffer = await fs.readFile(destFilePath);
+        let contents = buffer.toString('utf8');
+        if(contents.indexOf(rule.from) > -1) {
+            console.info(chalk.cyan(` -> Rewriting relative paths ${rule.from} => ${rule.to} for file "${filePath}"`));
+            contents = contents.replaceAll(rule.from, rule.to);
+        
             // Normalize transformed paths
+            const pathRegex = new RegExp(/((?:\.{1,2}[\/\\]{1,2})+[^'"\s]*)/gm);
             const matches = contents.match(pathRegex);
             if (matches && matches.length > 0) {
                 for (const match of matches) {
-                    // Generate the absolute path to the referenced match
-                    let refPath = path.resolve(destFolderPath, path.normalize(match))
-                    // Generate the relative path between the current processed file dir path & the referenced match path
-                    let relativePath = path.relative(destFolderPath, refPath)
-                    // Finally convert the path back to a POSIX compatible path
-                    relativePath = relativePath.split(path.sep).join(path.posix.sep)
-
-                    contents = contents.replaceAll(match, relativePath);
-
-                    if (this.options.debug) {
-                        console.log(chalk.grey(` -> Rewriting relative path ${match} => ${relativePath}`));
+                    if(match.indexOf(rule.to) > -1){
+                        // Generate the absolute path to the referenced match
+                        let refPath = path.resolve(destFolderPath, path.normalize(match))
+                        // Generate the relative path between the current processed file dir path & the referenced match path
+                        let relativePath = path.relative(destFolderPath, refPath)
+                        // Finally convert the path back to a POSIX compatible path
+                        relativePath = relativePath.split(path.sep).join(path.posix.sep)
+
+                        contents = contents.replaceAll(match, relativePath);
+
+                        if (this.options.debug) {
+                            console.log(chalk.grey(` -> Rewriting relative path ${match} => ${relativePath} in ${destFilePath}`));
+                        }
```

Comment:
Since we're now being more explicit in the replacement - do we still need this additional regex matching/replacement step?

Reply:
I think we do for situations where a replacement can be part of a larger string and includes multiple changes. We ran into an edge case. By using the regex we can extract the entire path and normalize on that, rather than normalizing the role.to property alone which can be skip normalizing an entire string if the replacement is part of a larger path.